velox/exec/Spiller.cpp

```diff
@@ -0,0 +1,402 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/exec/Spiller.h"
+
+#include "velox/common/base/AsyncSource.h"
+
+#include <folly/ScopeGuard.h>
+
+namespace facebook::velox::exec {
+
+void Spiller::extractSpill(folly::Range<char**> rows, RowVectorPtr* resultPtr) {
+  if (!*resultPtr) {
+    *resultPtr = std::static_pointer_cast<RowVector>(
+        BaseVector::create(rowType_, rows.size(), &spillPool()));
```

Comment:
nit: the pattern of calling BaseVector::create and casting to the right type repeats a lot in the codebase. It would be nice to add a template that would combine the two operations: `BaseVector::create<RowVector>(type, size, pool)`.